[DOCS-9128] NDM restructure (#26588)
DataDog/documentation: content/en/network_monitoring/devices/troubleshooting.md

Use the information below for troubleshooting Datadog Network Device Monitoring. If you need additional help, contact [Datadog support][1].

### Device not visible in Datadog

> The following assumes you are running Datadog Agent version `7.59.0+`.

Reviewer: Per discussion on slack, there are some nicer error messages in 7.61+

```suggestion
The following assumes you are running Datadog Agent version `7.61.0+`.
```

Author: Thanks @leeavital ! Per 2nd discussion, we'll keep as is for now and iterate on this and the Autodiscovery section once 7.61 is out.

   **Note**: Provide your credentials directly in the CLI. If credentials aren't provided, the Agent attempts to locate them in your running Agent configuration files. 

   **Linux**: 
   For SNMP v2:
   ```
   sudo -u dd-agent datadog-agent snmp walk <IP Address> -C <COMMUNITY_STRING>
   ```
   For SNMP v3:
    ```
    sudo -u dd-agent datadog-agent snmp walk <IP Address> -A <AUTH_KEY> -a <AUTH_PROTOCOL> -X <PRIV_KEY> -x <PRIV_PROTOCOL>
    ```
    **Windows**:   
    ```
    agent snmp walk <IP Address>[:Port]

    Example:           
    agent.exe snmp walk  10.143.50.30 1.3.6 
    ```

    Refer to your vendor specific documentation for additional information on running these commands.

3. If either the SNMP status or Agent walk show an error, it could indicate one of the following issues:

   **Unreachable or misconfigured device**:

   ```
   Error: check device reachable: failed: error reading from socket: read udp 127.0.0.1:46068->1.2.3.4:161
   ```
   
   **Solution**:
   - Log into your device and ensure that SNMP is enabled and exposed on port 161.

>    - Ensure you collector firewall allows egress:

Reviewer 2: ```suggestion
   - Ensure your collector firewall allows egress:
```

>    **Linux only**:

Reviewer 2: what parent section does this "Linux only" section belong to?  is it related to "Unreachable or misconfigured device" or is it its own issue?

in fact, I think this whole section could use more hierarchy—each solution should be a sub-section of its relevant issue

Author: it's related to the Unreachable section but yes.. Agree, let me rework this a bit!

   **Incorrect SNMPv2 credentials**:

   ``` 
   Error: an authentication method needs to be provided
   ```

   **Solution**: 
   If you are using SNMPv2 you need to set a community string.

   **Incorrect SNMPv3 privacy protocol**

   ```
   Error: check device reachable: failed: decryption error; failed to autodetect profile: failed to fetch sysobjectid: cannot get sysobjectid: decryption error; failed to fetch values: failed to fetch scalar oids with batching: failed to fetch scalar oids: fetch scalar: error getting oids `[1.3.6.1.2.1.1.1.0 1.3.6.1.2.1.1.2.0 1.3.6.1.2.1.1.3.0 1.3.6.1.2.1.1.5.0]`: decryption error  OR Error: check device reachable: failed: wrong digest; failed to autodetect profile: failed to fetch sysobjectid: cannot get sysobjectid: wrong digest; failed to fetch values: failed to fetch scalar oids with batching: failed to fetch scalar oids: fetch scalar: error getting oids `[1.3.6.1.2.1.1.1.0 1.3.6.1.2.1.1.2.0 1.3.6.1.2.1.1.3.0 1.3.6.1.2.1.1.5.0]`: wrong digest
   ```

   **Solution**: 
   Ensure the following configuration parameters are correct: `user`, `authKey`, `authProtocol`, `privKey`, `privProtocol`.

### Traps not being received for devices

> 1. Check the Datadog `agent.log` file to ensure that you aren't unable to bind to the traps port. If this is the case, the following error is present:

Reviewer 2: ```suggestion
Check the Datadog `agent.log` file to ensure that you can bind to the traps port. The following error indicates that you are unable to bind to the traps port:
```
this shouldn't be in the ordered list—the ordered list should just be the "solution" steps
also clearing up some unnecessarily convoluted language

   ```
   Failed to start snmp-traps server: error happened when listening for SNMP Traps: listen udp 0.0.0.0:162: bind: permission denied
   ```

   **Solution**:
   Add a net bind capability to the Agent binary, which allows the Agent to bind to reserved ports: 

   ```
   sudo setcap 'cap_net_bind_service=+ep' /opt/datadog-agent/bin/agent/agent
   ```

> 2. Navigate to the troubleshooting dashboard in NDM:

Reviewer 2: Wait, when do I move on to step 2? There was a solution in step 1, what if that solved my problem? Should I keep going?

Author: ahh yes that is confusing, will fix !

   {{< img src="/network_device_monitoring/troubleshooting/ndm_troubleshooting_dashboard.png" alt="The Network Device Monitoring page showing the Dashboard dropdown with the NDM Troubleshooting dashboard highlighed." style="width:80%;" >}}

3. Scroll down to the Traps widget and observe the **Traps incorrectly formatted** graph. If this is non-zero it likely means that the authentication on the NDM collector and the device do not match. 

   {{< img src="/network_device_monitoring/troubleshooting/ndm_traps_dashboard.png" alt="The NDM troubleshooting dashboard showing the Traps widget section." style="width:100%;" >}}
   

>    **Solution**: 

Reviewer 2: the problem/solution flow here just seems unintuitive

Author: Yes you're right! I reworked this.
